pkg/tests/tasks/traffic/ingress/gatewayapi_test.go

```diff
@@ -0,0 +1,164 @@
+package ingress
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/app"
+	"github.com/maistra/maistra-test-tool/pkg/tests/ossm"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	. "github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
+)
+
+func TestGatewayApiControllerMode(t *testing.T) {
+	NewTest(t).Id("T41").Groups(Full, InterOp).Run(func(t TestHelper) {
+		if env.GetSMCPVersion().LessThan(version.SMCP_2_3) {
+			t.Skip("TestGatewayApiControllerMode was added in v2.3")
+		}
+		ns := "foo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns)
+		})
+
+		ossm.DeployControlPlane(t)
+
+		smcpName := "basic"
+
+		t.LogStep("Install Gateway API CRD's")
+		shell.Executef(t, "kubectl get crd gateways.gateway.networking.k8s.io &> /dev/null ||   { kubectl kustomize github.com/kubernetes-sigs/gateway-api/config/crd/experimental?ref=v0.5.1 | kubectl apply -f -; }")
+
+		oc.CreateNamespace(t, ns)
+
+		t.LogStep("Install httpbin")
+		app.InstallAndWaitReady(t, app.Httpbin(ns))
+		t.Cleanup(func() {
+			app.Uninstall(t, app.Httpbin(ns))
+		})
+
+		t.LogStep("Deploy the Gateway app")
+		oc.ApplyString(t, ns, gatewayDeployment)
+		t.Cleanup(func() {
```

Comment:
You can group this inside the previous Cleanup, it's not necessary to to it after each statement

Reply:
No, doing this after each statement is actually better and correct. By using this pattern, there's less chance that we forget to clean up something. Also, if the test fails in the middle, the test only cleans up what it actually deployed instead of trying to clean up everything (if you do try to clean up everything, and an error occurs during cleanup, you might not clean up everything).

Reply:
updated in the latest commit.